catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -232,6 +233,90 @@ private Filter transformCustomFunctionToFilter(Filter filter) {
         return filter;
     }
 
+    private QueryRequest normalizeSort(QueryRequest request) {
+        if (request == null) {
+            return null;
+        }
+
+        Query query = request.getQuery();
+        if (query == null) {
+            return request;
+        }
+
+        List<SortBy> sortBys = new ArrayList<>();
+
+        // Primary sort parameter
+        SortBy sortBy = query.getSortBy();
+        SortBy newSortBy = normalizeSortBy(sortBy);
+        if (newSortBy != null) {
+            sortBys.add(newSortBy);
+        }
+
+        // Additional sort parameters
+        Serializable extraSortBys = request.getPropertyValue(EXT_SORT_BY);
+        if (extraSortBys != null && extraSortBys instanceof SortBy[]) {
+            List<SortBy> extraSortBysList = Arrays.asList((SortBy[]) extraSortBys);
+            extraSortBysList.stream()
+                    .map(this::normalizeSortBy)
+                    .filter(Objects::nonNull)
+                    .forEach(sortBys::add);
+        }
+
+        Map<String, Serializable> newProperties = request.getProperties();
+        SortBy normalizedSortBy = null;
+        if (sortBys.size() > 0) {
+            normalizedSortBy = sortBys.get(0);
+        }
+        if (sortBys.size() > 1) {
+            SortBy[] extraSortBysArray = sortBys.subList(1, sortBys.size())
+                    .toArray(new SortBy[0]);
+            newProperties.put(EXT_SORT_BY, extraSortBysArray);
+        } else {
+            newProperties.remove(EXT_SORT_BY);
+        }
+
+        Query newQuery = new QueryImpl(query,
+                query.getStartIndex(),
+                query.getPageSize(),
+                normalizedSortBy,
+                query.requestsTotalResultsCount(),
+                query.getTimeoutMillis());
+
+        return new QueryRequestImpl(newQuery,
+                request.isEnterprise(),
+                request.getSourceIds(),
+                newProperties);
+    }
+
+    private SortBy normalizeSortBy(SortBy cswSortBy) {
+        if (cswSortBy == null) {
+            return null;
+        }
+
+        if (cswSortBy.getPropertyName() != null
+                && !attributeRegistry.lookup(cswSortBy.getPropertyName()
+                .getPropertyName())
+                .isPresent() && !DefaultCswRecordMap.hasDefaultMetacardFieldForPrefixedString(
```

Comment:
This PR makes some changes to the way the DefaultCswRecordMap is used.  There may be some conflicts here https://github.com/codice/ddf/pull/2278

Reply:
It feels like tit shouldn't be hardcoded to use the defaultCswRecordMap at this level.  But instead, be able to substitute it for another map.  It appears the other PR @bdeining mentioned does that for the FilterVisitor and we would want to do something similar here.